executor/aggregate.go

```diff
@@ -226,11 +227,15 @@ func (d *HashAggIntermData) getPartialResultBatch(sc *stmtctx.StatementContext,
 // Close implements the Executor Close interface.
 func (e *HashAggExec) Close() error {
 	if e.isUnparallelExec {
-		e.memTracker.Consume(-e.childResult.MemoryUsage())
 		e.childResult = nil
 		e.groupSet, _ = set.NewStringSetWithMemoryUsage()
 		e.partialResultMap = nil
-		e.memTracker.ReplaceBytesUsed(0)
+		if e.memTracker != nil {
+			e.memTracker.ReplaceBytesUsed(0)
+		}
+		return e.baseExecutor.Close()
+	}
+	if !e.initialized {
```

Comment:
why add `if !e.initialized {` ? because the close should be called to close it child.

Reply:
we should only close channel when the channel is initialized, close a not initialized channel will cause panic, so if the executor is not initialized, we should return immediately.

Reply:
`e.baseExecutor.Close()` should close its child regardless this operator is compeletly initialized.

Reply:
In the end, L267, we will call e.baseExecutor.Close()

Reply:
plese refactor these code

Reply:
Why the `e.prepared` is not enough to control this?
I do not understand when will we close an un-initialized channel.

Reply:
@XuHuaiyu because e.prepared only protects if e.prepare4ParallelExec has been executed, but not if the channels are initialized.

> I do not understand when will we close an un-initialized channel.

Say, if the baseExecutor.Open() returns error, then `e.initForParallelExec(e.ctx)` will not be called, and when `e.Close()` is called, `e.prepared` will be false, and `close(e.inputCh)` will be executed, and will result in closing un-initialized channel.